### DISCUSSION TITLE
pylint init-hook yield to a crash

### DISCUSSION THREAD

**Discussion by toralf** [trunk-io/docs#303] [Feature Requests]
[pylint-crash-2024-06-30-18-31-47.txt](https://github.com/user-attachments/files/16045397/pylint-crash-2024-06-30-18-31-47.txt)

**Comment by gewenyu99** [comment]
Hey, thanks for the detailed report. I'll ping someone to take a look. 

Thanks!

**Comment by toralf** [comment]
Few more info:
```
tfoerste@p14s ~/devel/torutils $ env | grep PYTHONPATH
PYTHONPATH=/home/tfoerste/devel/stem
tfoerste@p14s ~/devel/torutils $ cat .pylintrc 
[MASTER]

disable=C0103,C0114,C0115,C0116,C0301,R0913,R0914,W0718,W0719

# low     Argument name "DBInstanceIdentifier" doesn't conform to snake_case naming style        pylint/C0103
# low     Missing module docstring                                                               pylint/C0114
# low     Missing class docstring                                                                pylint/C0115
# low     Missing function or method docstring                                                   pylint/C0116
# low     Line too long (173/100)                                                                pylint/C0301
# low     Too many arguments (6/5)                                                               pylint/R0913
# low     Too many local variables (21/15)                                                       pylint/R0914
# medium  Catching too general exception Exception                                               pylint/W0718
# medium  Raising too general exception: Exception                                               pylint/W0719

init-hook='import os; p = os.environ.get("PYTHONPATH"); sys.path.append(p)'

```

**Reply by gewenyu99** [reply]
Thanks for the extra context!

**Comment by ghost** [comment]
Wow, it works at work under WSL Ubuntu 22.04 Python 3.10. At home I do have a hardened Gentoo Linux with Python 3.12

**Reply by gewenyu99** [reply]
Yes it does :)

**Reply by toralf** [reply]
> Wow, it works at work under WSL Ubuntu 22.04 Python 3.10. At home I do have a hardened Gentoo Linux with Python 3.12

But other users have a mixed experiences.

**Comment by det** [comment]
Thanks for the report @toralf. I took a look, and it appears to be an internal Pylint crash. Are you able to run Pylint on your Project without trunk? You can test this either by installing a operating system package or by using a virtual env.

**Comment by toralf** [comment]
Running pylint under Gentoo Linux directly works fine. Running pylint via trunk.io (upgraded few minutes agot(pylint 3.2.4 → 3.2.5) yields to the crash. And it is reproducible (tested it here with any Python file from https://github.com/toralf/torutils):

```
init-hook='p = "/home/tfoerste/devel/stem"; sys.path.append(p); sys.path.append(".")'
#init-hook='import os; p = os.environ.get("PYTHONPATH"); sys.path.append(p)'

```
